src/CG/CG_elem.cpp

```diff
@@ -2511,11 +2528,25 @@ bool intersectPointSegment( array3D const &P, array3D const &P1, array3D const &
 {
     assert( validSegment(P1,P2) );
 
-    if(distancePointSegment(P, P1, P2) > distanceTolerance){
-        return false;
+    // Compute parameter t for the projection of P on the segment defined by P1 and P2
+    array3D n = P2 - P1;
+    double t = dotProduct(n, P - P1) / dotProduct(n, n);
+
+    // Check if the projection of P belongs to 
+    // the segment bounded by edges P1 and P2.
+    if (utils::DoubleFloatingGreater()(t, 0.0, distanceTolerance)
+    && utils::DoubleFloatingLess()(t, 1.0, distanceTolerance)) {
+
+        // Compute projection of P on the segment
+        array3D xP = (1. - t) * P1 + t * P2;
+
+        // Check if P lays on the line defined by the segment
+        if (norm2(P - xP) < distanceTolerance){
+            return true;
+        }
```

Comment:
In bitpit we have a function that projects a point to a segment, instead of re-writing such functionality you could just use the function we already have:
```
    assert( validSegment(P1,P2) );

    array3D xP     = projectPointSegment( P, P1, P2 );
    array3D offset = xP - P;

    return (dotProduct(offset, offset) < distanceTolerance * distanceTolerance);
```

Reply:
I cannot do that. I didn't use function projectPointSegment, because it enforces the projection point to belong into the segment. By using it, I cannot control if the given point actually belongs to the segment.

Reply:
It doesn't matter, if the point is not on the segment it will not coincide with its projection (neither the real projection nor the projection "moved" on the segment).

Reply:
Right! I did the modification.